jmh-core/src/main/java/org/openjdk/jmh/generators/core/MethodGroup.java

```diff
@@ -72,20 +72,19 @@ public int compareTo(MethodGroup o) {
 
     public void addMethod(MethodInfo method, int threads) {
         MethodInvocation mi = new MethodInvocation(method, threads);
-        MethodInvocation exist = methods.get(mi);
-        if (exist == null) {
-            methods.put(mi, mi);
-        } else {
+        if (methods.contains(mi)) {
```

Comment:
This could be simplified to `if (!methods.add(mi)) {`. This is a pretty idiom for adding the element once while testing for its existence.

Reply:
Good suggestion. Fixed.